diff --git a/node_modules/react-native-network-logger/src/Logger.ts b/node_modules/react-native-network-logger/src/Logger.ts
index ffe8dd9..a0115d6 100644
--- a/node_modules/react-native-network-logger/src/Logger.ts
+++ b/node_modules/react-native-network-logger/src/Logger.ts
@@ -1,4 +1,3 @@
-import XHRInterceptor from 'react-native/Libraries/Network/XHRInterceptor';
 import NetworkRequestInfo from './NetworkRequestInfo';
 import { Headers, RequestMethod, StartNetworkLoggingOptions } from './types';
 import extractHost from './utils/extractHost';
@@ -6,6 +5,12 @@ import { warn } from './utils/logger';
 import debounce from './utils/debounce';
 import { LOGGER_REFRESH_RATE, LOGGER_MAX_REQUESTS } from './constant';
 
+let XHRInterceptor: any;
+try {
+  XHRInterceptor = require('react-native/src/private/inspector/XHRInterceptor');
+} catch {
+  XHRInterceptor = require('react-native/Libraries/Network/XHRInterceptor');
+}
 let nextXHRId = 0;
 
 type XHR = {
